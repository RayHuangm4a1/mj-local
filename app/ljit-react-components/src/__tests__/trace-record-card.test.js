import React from 'react';
import { shallow, mount, } from 'enzyme';
import TraceRecordCard from '../components/trace-record-card';

describe('Trace Info Card', () => {
	it('handle default props', () => {
		const {
			onClick,
			traceRecord,
			className,
		} = TraceRecordCard.defaultProps;

		expect(onClick).toBeDefined();
		expect(onClick).toBeInstanceOf(Function);
		expect(traceRecord).toEqual({});
		expect(className).toBe('');
	});

	it('should renders correctly', () => {
		const wrapper = shallow(
			<TraceRecordCard
				traceRecord={{
					id: 201901240027,
					name: 'mock-name',
					lotteryName: 'mock-lottery',
					amountPerBet: 10,
					totalIssues: 2,
					reward: 0,
					issue: '20190701-0001',
					opencode: '',
					status: TraceRecordCard.StatusEnums.INCOMPLETE,
					createdAt: new Date('2019-04-13T10:38:43'),
				}}
				className='mock-class'
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-trace-info-card', () => {
		const wrapper = shallow(<TraceRecordCard/>);

		expect(wrapper.hasClass('ljit-trace-info-card')).toEqual(true);
	});

	it('should contains class ljit-trace-info-card__info', () => {
		const wrapper = shallow(<TraceRecordCard/>);

		expect(wrapper.find('.ljit-trace-info-card__info')).toHaveLength(1);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<TraceRecordCard className={className}/>);

		expect(wrapper.hasClass(className)).toEqual(true);
	});
	it('should handle onClick event', () => {
		const onClick = jest.fn();
		const wrapper = mount(
			<TraceRecordCard
				onClick={onClick}
			/>
		);

		wrapper.simulate('click');
		expect(onClick).toHaveBeenCalledTimes(1);
		expect(onClick).toHaveBeenCalled();
	});

	it('should mount in a full DOM', () => {
		const traceRecord = {
			id: 201901240027,
			name: 'mock-name',
			lotteryName: 'mock-lottery',
			amountPerBet: 10,
			totalIssues: 2,
			reward: 0,
			issue: '20190701-0001',
			opencode: '',
			status: TraceRecordCard.StatusEnums.INCOMPLETE,
			createdAt: new Date('2019-04-13T10:38:43'),
		};
		const className = 'mock-class';
		const onClick = () => {};
		const wrapper = mount(
			<TraceRecordCard
				traceRecord={traceRecord}
				className={className}
				onClick={onClick}
			/>
		);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().traceRecord).toMatchObject(traceRecord);
		expect(wrapper.props().onClick).toEqual(onClick);
	});
});
