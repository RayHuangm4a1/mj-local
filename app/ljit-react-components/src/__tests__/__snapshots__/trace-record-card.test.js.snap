// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trace Info Card should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TraceRecordCard
    className="mock-class"
    onClick={[Function]}
    traceRecord={
      Object {
        "amountPerBet": 10,
        "createdAt": 2019-04-13T02:38:43.000Z,
        "id": 201901240027,
        "issue": "20190701-0001",
        "lotteryName": "mock-lottery",
        "name": "mock-name",
        "opencode": "",
        "reward": 0,
        "status": 1,
        "totalIssues": 2,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <InfoCard
        className="ljit-trace-info-card__info"
        isShowingCloseButton={false}
        left={
          <React.Fragment>
            <LabelText
              fontSize="small"
              label="方案号"
              labelColType="medium"
              labelStyle={Object {}}
              text={201901240027}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="玩法"
              labelColType="medium"
              labelStyle={Object {}}
              text="mock-name"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="单注金额"
              labelColType="large"
              labelStyle={Object {}}
              text={10}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="中奖金额"
              labelColType="large"
              labelStyle={Object {}}
              text="-"
              textStyle={Object {}}
            />
          </React.Fragment>
        }
        onClose={[Function]}
        right={
          <React.Fragment>
            <LabelText
              fontSize="small"
              label="期号"
              labelColType="small"
              labelStyle={Object {}}
              text="20190701-0001"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="时间"
              labelColType="small"
              labelStyle={Object {}}
              text="4/13 10:38:43"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="追号期数"
              labelColType="large"
              labelStyle={Object {}}
              text={2}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="中奖号码"
              labelColType="large"
              labelStyle={Object {}}
              text="-"
              textStyle={Object {}}
            />
          </React.Fragment>
        }
        topLeft={
          <div
            className="ljit-trace-info-card__header"
          >
            <span
              className="ljit-trace-info-card__title"
            >
              mock-lottery
            </span>
            <StatusTag
              backgroundColor=""
              status="traceIncomplete"
              tagType="default"
              text="未完成"
              textColor=""
            />
          </div>
        }
      />,
      "className": "ljit-trace-info-card mock-class",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "ljit-trace-info-card__info",
        "isShowingCloseButton": false,
        "left": <React.Fragment>
          <LabelText
            fontSize="small"
            label="方案号"
            labelColType="medium"
            labelStyle={Object {}}
            text={201901240027}
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="玩法"
            labelColType="medium"
            labelStyle={Object {}}
            text="mock-name"
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="单注金额"
            labelColType="large"
            labelStyle={Object {}}
            text={10}
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="中奖金额"
            labelColType="large"
            labelStyle={Object {}}
            text="-"
            textStyle={Object {}}
          />
        </React.Fragment>,
        "onClose": [Function],
        "right": <React.Fragment>
          <LabelText
            fontSize="small"
            label="期号"
            labelColType="small"
            labelStyle={Object {}}
            text="20190701-0001"
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="时间"
            labelColType="small"
            labelStyle={Object {}}
            text="4/13 10:38:43"
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="追号期数"
            labelColType="large"
            labelStyle={Object {}}
            text={2}
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="中奖号码"
            labelColType="large"
            labelStyle={Object {}}
            text="-"
            textStyle={Object {}}
          />
        </React.Fragment>,
        "topLeft": <div
          className="ljit-trace-info-card__header"
        >
          <span
            className="ljit-trace-info-card__title"
          >
            mock-lottery
          </span>
          <StatusTag
            backgroundColor=""
            status="traceIncomplete"
            tagType="default"
            text="未完成"
            textColor=""
          />
        </div>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <InfoCard
          className="ljit-trace-info-card__info"
          isShowingCloseButton={false}
          left={
            <React.Fragment>
              <LabelText
                fontSize="small"
                label="方案号"
                labelColType="medium"
                labelStyle={Object {}}
                text={201901240027}
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="玩法"
                labelColType="medium"
                labelStyle={Object {}}
                text="mock-name"
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="单注金额"
                labelColType="large"
                labelStyle={Object {}}
                text={10}
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="中奖金额"
                labelColType="large"
                labelStyle={Object {}}
                text="-"
                textStyle={Object {}}
              />
            </React.Fragment>
          }
          onClose={[Function]}
          right={
            <React.Fragment>
              <LabelText
                fontSize="small"
                label="期号"
                labelColType="small"
                labelStyle={Object {}}
                text="20190701-0001"
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="时间"
                labelColType="small"
                labelStyle={Object {}}
                text="4/13 10:38:43"
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="追号期数"
                labelColType="large"
                labelStyle={Object {}}
                text={2}
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="中奖号码"
                labelColType="large"
                labelStyle={Object {}}
                text="-"
                textStyle={Object {}}
              />
            </React.Fragment>
          }
          topLeft={
            <div
              className="ljit-trace-info-card__header"
            >
              <span
                className="ljit-trace-info-card__title"
              >
                mock-lottery
              </span>
              <StatusTag
                backgroundColor=""
                status="traceIncomplete"
                tagType="default"
                text="未完成"
                textColor=""
              />
            </div>
          }
        />,
        "className": "ljit-trace-info-card mock-class",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "ljit-trace-info-card__info",
          "isShowingCloseButton": false,
          "left": <React.Fragment>
            <LabelText
              fontSize="small"
              label="方案号"
              labelColType="medium"
              labelStyle={Object {}}
              text={201901240027}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="玩法"
              labelColType="medium"
              labelStyle={Object {}}
              text="mock-name"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="单注金额"
              labelColType="large"
              labelStyle={Object {}}
              text={10}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="中奖金额"
              labelColType="large"
              labelStyle={Object {}}
              text="-"
              textStyle={Object {}}
            />
          </React.Fragment>,
          "onClose": [Function],
          "right": <React.Fragment>
            <LabelText
              fontSize="small"
              label="期号"
              labelColType="small"
              labelStyle={Object {}}
              text="20190701-0001"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="时间"
              labelColType="small"
              labelStyle={Object {}}
              text="4/13 10:38:43"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="追号期数"
              labelColType="large"
              labelStyle={Object {}}
              text={2}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="中奖号码"
              labelColType="large"
              labelStyle={Object {}}
              text="-"
              textStyle={Object {}}
            />
          </React.Fragment>,
          "topLeft": <div
            className="ljit-trace-info-card__header"
          >
            <span
              className="ljit-trace-info-card__title"
            >
              mock-lottery
            </span>
            <StatusTag
              backgroundColor=""
              status="traceIncomplete"
              tagType="default"
              text="未完成"
              textColor=""
            />
          </div>,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
